### PR TITLE
shot: fix headers, minor cleanup

### LINKED ISSUE #25187: shot `Headers` index signature should be `string | string[]`
If you know how to fix the issue, make a pull request instead.

- [x] I tried using the `@types/shot` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [x] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @AJamesPhillips

See title, if one header is sent multiple times the index value is an array.

### PR DESCRIPTION
**Pull request by SimonSchick**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: This is not explicitly documented but is current behaviour as this is proxied to `http.IncomingMessage`(node)
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

Fixes #25187

### CODE REVIEW

**Comment by SimonSchick** [comment]
cc @lenovouser @jhsimms since this is hapi ecosys

**Review by jhsimms — APPROVED** [review]

**Review by lenovouser — APPROVED** [review]
👌🏻

File: types/shot/index.d.ts [hunk 1]
```diff
@@ -1,115 +1,111 @@
 // Type definitions for shot 3.4
 // Project: https://github.com/hapijs/shot
-// Definitions by: AJP <https://github.com/AJamesPhillips>
+// Definitions by:  AJP <https://github.com/AJamesPhillips>
+//                  Simon Schick <https://github.com/SimonSchick>
```
Comment (lenovouser) [inline types/shot/index.d.ts]: One space too much

**Comment by SimonSchick** [comment]
Hey @emcasay since you are merging stuff rn :)

**Comment by amcasey** [comment]
@SimonSchick I'm assuming you meant me. :smile:

**Pull request merged**